### PR TITLE
Consolidating stack

### PR DESCRIPTION
**Pull request by Kelym**
Clean up demo_interface and addon functions.
Leftover TODOs for cleaning up the stack (ETA Feb 2024):
[ ] separate certain functions to tycho_infra
[ ] nn_backlash
[ ] camera support

### CODE REVIEW

**Comment by Kelym** [comment]
@abhaybd could you take a look at this?
[ ] could you help me ensure that [this commit](https://github.com/personalrobotics/tycho_demo/pull/3/commits/f0b5d28918cc35c97b7e6e6a3c26f1ebceb06dab) looks good?
[ ] can we remove redundant replay functions in addon/?

File: tycho_demo/addon/realsense.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+import numpy as np
+import cv2
+import pyrealsense2 as rs
+from PIL import Image
+from threading import Thread
+import queue
+from multiprocessing import Queue as MPQueue
+from multiprocessing import Process
+
+from franka_demo.utils import print_and_cr
```
Comment (abhaybd) [inline tycho_demo/addon/realsense.py]: Should that be `tycho_demo`?

File: tycho_demo/addon/recording.py [hunk 2]
```diff
@@ -71,30 +67,33 @@ def _count_recording(key, state):
...
+    label_failure_demo(state.last_rosbag, state.ros_record_dicts)
 
 # ------------------------------------------------------------------------------
 
-def start_rosbag_recording(record_prefix, pose_topics, cameras=DEFAULT_CAMERAS):
+def start_rosbag_recording(record_prefix, pose_topics, dict_topics):
   print_and_cr(colors.bg.green + 'Recording to rosbag {}'.format(
     os.path.basename(record_prefix)))
-  args1 = ['rosbag', 'record'] + pose_topics + \
+  args1 = ['rosbag', 'record'] + pose_topics[:-1] + \
```
Comment (abhaybd) [inline tycho_demo/addon/recording.py]: Why are we excluding the last topic?

File: tycho_demo/addon/recording.py [hunk 3]
```diff
@@ -103,17 +102,43 @@ def stop_rosbag_recording(cameras=DEFAULT_CAMERAS):
...
+    f'{rosbag_recording_to}-{_topic}.bag' for _topic in dict_topics.keys()]
+  for fn in list_fn:
+    while not os.path.isfile(fn):
+      sleep(0.05)
+    split_fn = fn.split('/')
+    new_fn = split_fn[-1].split('-')
+    new_fn[-1] = 'fail-'+new_fn[-1]
+    split_fn[-1] = '-'.join(new_fn)
+    cmd = "mv {} {}".format(fn, '/'.join(split_fn))
+    os.system(cmd)
```
Comment (abhaybd) [inline tycho_demo/addon/recording.py:130]: Should this be in a separate thread so we don't block the control thread?

File: tycho_demo/addon/snapping.py [hunk 4]
```diff
@@ -36,45 +37,65 @@ def do_snapping(state, moving_positions, total_time, return_mode=None):
...
+def _flat_move(key, state):
+  import numpy as np
+  from tycho_env.utils import construct_choppose, construct_command
+  from scipy.spatial.transform import Rotation as scipyR
   print_and_cr('Move to a predefined sets of positions')
+  moving_eepos = np.empty(8)
+  moving_eepos[:3] = [-0.22083452, -0.38280237, 0.10270547]
+  moving_eepos[3:7] = scipyR.from_rotvec([0., 180, 0.], degrees=True).as_quat()
+  moving_eepos[7] = -0.437
+  jointpos = construct_command(state.arm, state.current_position, target_vector=moving_eepos)
```
Comment (abhaybd) [inline tycho_demo/addon/snapping.py:49]: This is my bad, we should probably lock the state lock here

**Review by Kelym — COMMENTED** [review]
Stack merged to latest. Noticeable changes: add support for az camera; consolidated infra (a few TODOs remain as described in the PR).

**Review by abhaybd — APPROVED** [review]

**Pull request merged**